module4/ex14_thresholdequal/ex14_thresholdequal.py

```diff
@@ -1,5 +1,38 @@
 class ThresholdEqual:
-    pass
+    def __init__(self, x):
+        self.x = x
+        self.tolerance = 2
+
+    def __eq__(self, other):
+        return abs(self.x - other.x) <= self.tolerance
+
+    def __lt__(self, other):
+        return self.x - other.x < 0 and abs(self.x - other.x) > self.tolerance
+
+    def __gt__(self, other):
+        return self.x - other.x > 0 and abs(self.x - other.x) > self.tolerance
+
+    def __add__(self, other):
+        return self.x + other.x
+
+    def __iadd__(self, other):
+        self.x += other.x
+        return self
```

Comment:
Nothing to change here, just some mentions. 
- Saw indeed that tests do not pass if returning type for `__add__` is not a number but a new Threshold class. Despite this, returning a result of the same type as the operands themselves is the way operators should work. Here or the tests should be changed to consider this or for tests to have passed it should have to rewrite` __eq__`  so it compares Threshold objects also to numbers, but that would imply for consistency to implement the other operators with `other` being a number and it would have been too complicated.
- Nice that `__iadd__` operator is implemented as an in-place operation and return the object itself not a new one.
- For validation it could be validated that `x` is a number in init and `other` is a Threshold object (or a number).